src/uu/ls/src/url_escaper.rs

```diff
@@ -0,0 +1,147 @@
+use std::{borrow::Cow, convert::{TryFrom, TryInto}, ffi::OsStr, ptr, mem::ManuallyDrop};
+
+mod byte_or_escape
+{
+    #[derive(Copy, Clone, PartialEq, Eq)]
+    #[repr(transparent)]
+    pub struct ByteOrEscape(u16);
+
+    impl ByteOrEscape {
+        pub fn byte(b: u8) -> Self {
+            Self(0x8000u16 | (b as u16))
+        }
+
+        pub fn identity(b: u8) -> Self {
+            Self::byte(b)
+        }
+
+        pub fn replacement(b: u8) -> Self {
+            Self::byte(b)
+        }
```

Comment:
What is the difference between these two?

Reply:
None, it's there to support changing the algorithm more easily.

Reply:
Then I think it can be removed for now.